src/v/cloud_storage/remote.cc

```diff
@@ -564,14 +564,19 @@ ss::future<download_result> remote::download_segment(
     retry_chain_node fib(&parent);
     retry_chain_logger ctxlog(cst_log, fib);
     auto path = cloud_storage_clients::object_key(segment_path());
+
+    auto m = _probe.client_acquisition();
     auto lease = co_await _pool.local().acquire(fib.root_abort_source());
+    m.reset();
```

Comment:
This reset doesn't look very good but using RAII here would require adding a scope, initializing lease early etc.